lib/puppet/pops/parser/evaluating_parser.rb

```diff
@@ -24,20 +23,19 @@ def parse_string(s, file_source = nil)
     # Also a possible improvement (if the YAML parser returns positions) is to provide correct output of position.
     #
     begin
-      assert_and_report(parser.parse_string(s, file_source)).model
+      assert_and_report(parser.parse_string(s, file_source), file_source).model
     rescue Puppet::ParseErrorWithIssue => e
       raise e
     rescue Puppet::ParseError => e
       # TODO: This is not quite right, why does not the exception have the correct file?
-      e.file = @file_source unless e.file.is_a?(String) && !e.file.empty?
+      e.file = file_source unless e.file.is_a?(String) && !e.file.empty?
       raise e
     end
   end
 
   def parse_file(file)
```

Comment:
Looks like this also needs the new arg passed.

Reply:
shouldn't need to be passed, just a slight error in the way i rewrote the assert_and_report call